showing promotion list

iamcxa opened the pull request:
- change 
  - routing changed for PromotionController
  - shop-discount-detail.jade changed.
  - model Promotion changed
- add
  - added PromotionController.js
  - added PromotionService.js
- function ok
  - now list all promotions works.

iamcxa committed:
add promotion pre-built data.

iamcxa committed:
change PromotionController routing

iamcxa committed:
add PromotionController, now list shows.

iamcxa committed:
update create/update/delete method for PromotionController API.

iamcxa committed:
update try/cacth

iamcxa committed:
add discount.js for view.

iamcxa committed:
pre-built ajax function for click add a 閃購 button

smlsunxie commented on api/controllers/PromotionController.js:
```diff
@@ -0,0 +1,66 @@
+  list: async (req, res) => {
+    try {
+      let promotions = await PromotionService.findAll();
+      return res.view('user/controlShopDiscount', {
```
移到 view/promotion/controlShopDiscount 底下

後面的都是喔

smlsunxie commented:
日期格式 format 一下喔

可以問一下耀德

<img width="596" alt="screen shot 2015-09-09 at 8 41 13 pm" src="https://cloud.githubusercontent.com/assets/1689227/9762027/2e718078-5733-11e5-93f5-e13524ae46d6.png">

iamcxa committed:
1. Merge remote-tracking branch 'upstream/develop' into develop
2. move promotion views from /user into /promotion folder.
3. fix promotion date display.
4. change model Promotion field 'type'.
5. change pre-built promotion data.

Conflicts:
	config/init/index.js

iamcxa committed:
add pre-built data field type.

iamcxa committed:
Merge remote-tracking branch 'upstream/develop' into develop

Conflicts:
	config/init/index.js
	views/_layout_admin.jade
	views/layout.jade
